core/translate/where_clause.rs

```diff
@@ -173,49 +187,71 @@ pub fn translate_processed_where<'a>(
             continue;
         }
         let target_jump = current_loop.next_row_label;
-        translate_condition_expr(program, select, &term.expr, target_jump, false, cursor_hint)?;
+        let success_label = program.allocate_label();
+        translate_condition_expr(
+            program,
+            select,
+            &term.expr,
+            cursor_hint,
+            ConditionMetadata {
+                jump_on_success: false,
+                on_failure_jump_label: target_jump,
+                on_success_jump_label: success_label,
+            },
+        )?;
+
+        program.resolve_label(success_label, program.offset());
     }
 
     Ok(())
 }
 
+#[derive(Default, Debug, Clone, Copy)]
+struct ConditionMetadata {
```

Comment:
btw do we want to care about alignment at all when using stuff like this, or is it just dumb micro-optimization? this becomes 24 bytes due to the i64 BranchOffsets, but realistically we _could_ use smaller integers for a bunch of stuff in our code

Reply:
Larger struct size means more memory allocated, which means CPU cycles wasted. So absolutely makes sense to think about struct layout!